lib/miq_automation_engine/engine/miq_ae_builtin_method.rb

```diff
@@ -47,11 +57,12 @@ def self.powershell(_obj, inputs)
 
     def self.miq_parse_provider_category(obj, _inputs)
       provider_category = nil
-      keys = %w(vm orchestration_stack miq_request miq_provision miq_host_provision vm_migrate_task platform_category)
-      keys.detect { |k| provider_category = category_for_key(obj, k) }
+      ATTRIBUTE_LIST.detect { |attr| provider_category = category_for_key(obj, attr) }
```

Comment:
@gmcculloug 
We are traversing the ATTRIBUTE_LIST and detect would return the first attribute name that matches the criteria. In this case we are calling another function, that takes the attribute and returns something different from the attribute name. So I think just doing the detect and doing the assignment will give us the attribute name not the provider_category.